tests/teststun.nim

```diff
@@ -0,0 +1,87 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.used.}
+
+import options, strutils
+import bearssl
+import ../webrtc/udp_connection
+import ../webrtc/stun/stun_connection
+import ../webrtc/stun/stun_message
+import ../webrtc/stun/stun_attributes
+import ../webrtc/stun/stun_utils
+import ./asyncunit
+
+proc newRng(): ref HmacDrbgContext =
+  HmacDrbgContext.new()
+
+suite "Stun message encoding/decoding":
+  test "Get BindingRequest + encode & decode with a set username":
+    let
+      udpConn = UdpConn.init(AnyAddress)
+      conn = StunConn.init(udpConn, TransportAddress(AnyAddress), true, newRng())
+      msg = conn.getBindingRequest(username = "Test")
+      encoded = msg.encode(msg.getAttribute(AttrUsername))
+      decoded = StunMessage.decode(encoded)
+
+    check:
+      msg.msgType == decoded.msgType
+      msg.transactionId == decoded.transactionId
+      msg.getAttribute(AttrUsername) == decoded.getAttribute(AttrUsername)
+      msg.getAttribute(AttrICEControlling).isSome()
+      msg.getAttribute(AttrICEControlled).isNone()
```

Comment:
This doesn't seem to be related to encoding/decoding, does it?

Reply:
I should have tested `decoded` instead, you're right.
I fixed it.